lib/ansible/vars/manager.py

```diff
@@ -206,20 +206,18 @@ def get_vars(self, play=None, host=None, task=None, include_hostvars=True, inclu
 
             def _get_plugin_vars(plugin, path, entities):
                 data = {}
-                try:
+                if hasattr(plugin, 'get_vars'):
```

Comment:
This is not [duck typing](https://docs.python.org/2/glossary.html#term-duck-typing), and is actually considered an anti pattern in Python.  In python the general rule is [EAFP](https://docs.python.org/2/glossary.html#term-eafp)

Catching the `AttributeError` is the preferred means of achieving this behavior.

Reply:
Ok, adding orig_exc=e to the AnsibleError would achieve what I'm looking for so I'll revert this and use that.

Reply:
@sivel I can't include both exceptions in the AnsibleError and since I don't know which interface is used is it ok if I add a debug or vvv display with the first AttributeError. Or are there other ways to make the effective error visible ?

Reply:
I don't think I would have a problem with adding that information to `-vvv`.  That is fairly typical style.

Reply:
that is what orig_exc basically does, chaining it to any other tb in chain